### PR TITLE
fix: folder name leaked in Datadog logs WPB-1152

### PR DESCRIPTION
**Pull request by johnranj**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1152" title="WPB-1152" target="_blank">StoryWPB-1152</a>  [iOS] DataDog integration leaks Conversation Folder Names
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Conversation Folder Names are leaked on datadog integration

### Solutions

Updated alias name for conversation folder.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift [hunk 1]
```diff
@@ -73,8 +73,8 @@ final class ConversationListViewModel: NSObject {
                     return "groups"
                 case .favorites:
                     return "favorites"
-                case .folder(label: let label):
-                    return label.name ?? "folder"
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift]: @johnranj  I don't really get how this avoids the info to be sent to datadog?

Comment (johnranj) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift]: label.name which provides the folder name. Here the alias for the folder is just "folder" not the actual folder name.
Do you have any other suggestion for this ?

Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift]: the goal was **not to send** the value, it seems here you unwrap this value if present. Or am I missing something?

Comment (johnranj) [reply wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift]: Yes, the goal is not to send the value. But we don't have control towards field level restriction. We have only object level restriction. Alternate fix could be , returning a generic alias like other labels like favourites, groups etc.

File: wire-ios/Wire-iOS/Sources/Managers/DatadogWrapper.swift [hunk 2]
```diff
@@ -78,6 +78,12 @@ public class DatadogWrapper {
                 .trackBackgroundEvents()
                 .trackRUMLongTasks()
                 .enableCrashReporting(using: DDCrashReportingPlugin())
+                // Redacting an event information on tap action 
+                .setRUMActionEventMapper { actionEvent in
+                    var actionEvent = actionEvent
+                    actionEvent.action.target?.name = "ConversationListCell"
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Managers/DatadogWrapper.swift]: don't you redact all actions here, even if there are not from ConversationListCell ?

Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/Managers/DatadogWrapper.swift]: deleted

**Review by agisilaos — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**